### ISSUE TITLE
load page only when index is selected on bottomNavBar

### ISSUE THREAD

**Issue by Ansh-Rathod** [flutter/flutter#108417]
Hi,  this would be easy to understand if you see both videos. first, on the left-sided video, I used the normal Cupertino tab view which I wanted but I'm facing issues with it so I changed my bottom navbar with this different tab navigator(right-sided video) but I want to keep this right-sided tab view with navigator but the problem is in the right video is, it loads all pages while opening the app and I don't want that which is slowing frames u can feel it in CircularProgressIndicator.
the issue is how can achieve this like load the only first page and the rest of the ones on the tabselected(like left-sided video in the second tab navigator code).

<table>
  <tr>
    <td>i want this with diffrent code</td>
     <td>i don't want </td>
  </tr>
  <tr>
    <td><video src="https://user-images.githubusercontent.com/67627096/181173263-d06bd3ad-180e-4089-abbf-702179c5ceca.mp4"></td>
    <td><video src="https://user-images.githubusercontent.com/67627096/181173172-501fd87b-ffb5-4cb5-bd75-c5630b7c2409.mp4"></td>
    
  </tr>
 </table>

code of the first video
```
 CupertinoTabScaffold(
      restorationId: 'CINEASTE',
      tabBar: CupertinoTabBar(
        border: Border(
          top: BorderSide(
            color: Theme.of(context).dividerColor,
            width: .2,
          ),
        ),
        backgroundColor: Theme.of(context).scaffoldBackgroundColor,
        onTap: (index) {
          setState(() {
            currentIndex = index;
          });
        },
        inactiveColor: Theme.of(context).bottomAppBarColor,
        activeColor: Theme.of(context).unselectedWidgetColor.withOpacity(1),
        currentIndex: currentIndex,
        iconSize: 24.0,
        items: const [
          BottomNavigationBarItem(
            activeIcon: Icon(CustomIcons.home_filled),
            icon: Icon(CustomIcons.home),
          ),
          BottomNavigationBarItem(
            activeIcon: Icon(CustomIcons.search_filled),
            icon: Icon(CustomIcons.search),
          ),
          BottomNavigationBarItem(
            activeIcon: Icon(CustomIcons.category_filled),
            icon: Icon(CustomIcons.category),
          ),
          BottomNavigationBarItem(
            activeIcon: Icon(CustomIcons.profile_filled),
            icon: Icon(CustomIcons.profile),
          ),
        ],
      ),
      tabBuilder: (context, index) {
        return buildCurrentPage(index);
      },
```

code of the second video

```

class BottomAppBarTwoState extends State<BottomAppBarTwo> {
  String _currentPage = "home";
  List<String> pageKeys = ["home", "trending", "explore", "profile"];

  FocusNode homeNode = FocusNode(debugLabel: 'home');
  FocusNode profileNode = FocusNode(debugLabel: 'profile');
  FocusNode exploreNode = FocusNode(debugLabel: 'explore');
  FocusNode trendingNode = FocusNode(debugLabel: 'trending');

  List<FocusNode> nodeList = [];

  final Map<String, GlobalKey<NavigatorState>> _navigatorKeys = {
    "home": GlobalKey<NavigatorState>(),
    "trending": GlobalKey<NavigatorState>(),
    "explore": GlobalKey<NavigatorState>(),
    "profile": GlobalKey<NavigatorState>(),
  };
  int _selectedIndex = 0;

  void _selectTab(String tabItem, int index) {
    if (tabItem == _currentPage) {
      _navigatorKeys[tabItem]!.currentState!.popUntil((route) => route.isFirst);
    } else {
      setState(() {
        _currentPage = pageKeys[index];
        _selectedIndex = index;
      });
    }
  }

  var currentUser = Hive.box('settings').get('currentuser');
  @override
  void initState() {
    super.initState();

    nodeList = [homeNode, trendingNode, exploreNode, profileNode];
    
  }

  @override
  Widget build(BuildContext context) {
    return WillPopScope(
        onWillPop: () async {
          final isFirstRouteInCurrentTab =
              !await _navigatorKeys[_currentPage]!.currentState!.maybePop();
          if (isFirstRouteInCurrentTab) {
            if (_currentPage != "home") {
              _selectTab("home", 1);
              return false;
            }
          }
          return isFirstRouteInCurrentTab;
        },
        child: Scaffold(
          resizeToAvoidBottomInset: true,
          body: Stack(
            children: <Widget>[
              _buildOffstageNavigator("home"),
              _buildOffstageNavigator("trending"),
              _buildOffstageNavigator("explore"),
              _buildOffstageNavigator("profile"),
            ],
          ),
          bottomNavigationBar: CustomCupertinoTabBar(
            backgroundColor: Theme.of(context).scaffoldBackgroundColor,
            inactiveColor: Theme.of(context).bottomAppBarColor,
            activeColor: Theme.of(context).unselectedWidgetColor.withOpacity(1),
            iconSize: 24.0,
            onTap: (int index) {
              _selectTab(pageKeys[index], index);
            },
            currentIndex: _selectedIndex,
            items: const [
              BottomNavigationBarItem(
                activeIcon: Icon(CustomIcons.home_filled),
                icon: Icon(CustomIcons.home),
              ),
              BottomNavigationBarItem(
                activeIcon: Icon(CustomIcons.search_filled),
                icon: Icon(CustomIcons.search),
              ),
              BottomNavigationBarItem(
                activeIcon: Icon(CustomIcons.category_filled),
                icon: Icon(CustomIcons.category),
              ),
              BottomNavigationBarItem(
                activeIcon: Icon(CustomIcons.profile_filled),
                icon: Icon(CustomIcons.profile),
              ),
            ],
          ),
        ));
  }

  Widget _buildOffstageNavigator(String tabItem) {
    return HeroMode(
      enabled: _currentPage == tabItem,
      child: Offstage(
        offstage: _currentPage != tabItem,
        child: FocusScope(
          canRequestFocus: _currentPage == tabItem,
          child: TabNavigator(
            controller: widget.con,
            currentUser: currentUser,
            navigatorKey: _navigatorKeys[tabItem]!,
            tabItem: tabItem,
          ),
        ),
      ),
    );
  }
}

class TabNavigator extends StatelessWidget {
  final CurrentUserModel currentUser;
  final SettingsController controller;
  const TabNavigator({
    Key? key,
    required this.currentUser,
    required this.controller,
    required this.navigatorKey,
    required this.tabItem,
  }) : super(key: key);
  final GlobalKey<NavigatorState> navigatorKey;
  final String tabItem;

  @override
  Widget build(BuildContext context) {
    Widget child = Container();

    if (tabItem == "home") {
      child = const FeedScreen();
    } else if (tabItem == "trending") {
      child = const TrendingScreen();
    } else if (tabItem == "explore") {
      child = const ExplorePage();
    } else if (tabItem == "profile") {
      child = CurrentProfileScreen(
        con: controller,
        username: currentUser.username,
      );
    }
    return Navigator(
      key: navigatorKey,
      onGenerateRoute: (routeSettings) {
        return createRoute(child);
      },
    );
  }
}
```

Basically i want same result of the first video by changing something in code of the second code. 

Thanks!

**Comment by superciccio** [comment]
@Ansh-Rathod I guess your problem is here 

`
Stack(
            children: <Widget>[
              _buildOffstageNavigator("home"),
              _buildOffstageNavigator("trending"),
              _buildOffstageNavigator("explore"),
              _buildOffstageNavigator("profile"),
            ],
          ),
`


I have something in one of my project, where I basically build the screen from an index (botttomNavigationBar).
I hope you get my hint...
